assignment4.Rmd

```diff
@@ -105,14 +147,144 @@ sweep(loadings, 2, colSums(loadings), "/") #sweep() computes each row as a propo
+
+This PC gives a lot of weight to social sciences and leadership-related (including school administration) programs. Every single subject among the 10 with largest positive weight falls into such areas. Interestigly, this dimension has nothing to do with our initial exploratory analysis!(But hold on, Component 3 does have something to do with it!)
+
+
+```{r}
+
+names(TCpca$rotation[order(-TCpca$rotation[,1]),1][1:10])
+
+```
+
+## PC2 - Psychology-and-Health dimension (8.1% of variance)
+
+We print below the 10 programs with largest positive weight for PC2.
+
+This gives a lot of weight to programs related to health and psychology. Every very single subject among the 10 with largest positive weight falls into such category - except for the 10th largest, "Physical Education" (this is not too fal from the "Heath" topic, though - is it?). 
+
+```{r}
+# 10 programs with largest positive weight for PC2.
+
+names(TCpca$rotation[order(-TCpca$rotation[,2]),2][1:10])
+
+```
+
+## PC3 - The-future-of-Ed-dimension (6.2% of variance) 
+We print below the 10 programs with largest positive weight for PC3.
+
+We finally found something that has some relationship with our initial exploratory analysis. This dimension includes in its 10 most relevant positive variables almost exclusively technology-related programs. The only exceptions here are "cognitive science" and "mathematics". 
+
+```{r}
+# 10 programs with largest positive weight for PC3.
+
+names(TCpca$rotation[order(-TCpca$rotation[,3]),3][1:10])
+
+```
+
+## PC4 - The non-teacher-education-dimension (4.9% of variace)
+ 
+This componet seemed quite hard to interpretate, as it shows a combination of elements related to the dimensions PC1 and PC2 such as  "Leadership", "Sociology", "Politics", "Change.Leadership", "Physiology", "Economics.and.Education", "Health.Education", and"Nursing". In this case, I did get curious to see what were the largest negative weights. Interestingly, some coherence around the topic of "tecaher education" shows up. This is the case for "English.Education", "Teaching.English", "Literacy", "Bilingual.Bicultural.Education", "Reading", and "Early.Childhood.Special.Education". So I suggest that this dimension might be interpreted as the "non-teacher-education" dimension! To me, this actually makes some sense once we remember that this data is from Teachers College. At this school Teacher Ed. vs. Non-Techer Ed. is indeed a meaningful divider.
+
+We print below the 10 programs with largest positive weight for PC4. We also do the same for the 10 larges negative weights.  
+
+```{r}
+# 10 programs with largest positive weight for PC4.
+names(TCpca$rotation[order(-TCpca$rotation[,4]),4][1:10])
+
+#10 programs with largest NEGATIVE weight for PC4.
+names(TCpca$rotation[order(TCpca$rotation[,4]),4][1:10])
+
```

Comment:
Overall, excellent job! Keep up the good work!

Reply:
Thank you , Anna! :D